tests/js/object-tests.js

```diff
@@ -515,5 +515,30 @@ module.exports = {
         TestCase.assertEqual(realm.objects('Date')[1].currentDate.getTime(), -10000);
         TestCase.assertEqual(realm.objects('Date')[2].currentDate.getTime(), 1000000000000);
         TestCase.assertEqual(realm.objects('Date')[3].currentDate.getTime(), -1000000000000);
+    },
+    
+    testInheritance: function() {
+        // the equivalent of `class InheritingObject extends Realm.Object`
+        function InheritingObject() {
```

Comment:
@fealebenpae @alazier FYI, native ES6 classes will have issues in Node, and I think fixing it will require removing the "feature" where the constructor is automatically called.